url-dsl/src/main/scala/urldsl/language/PathSegmentImpl.scala

```diff
@@ -28,8 +28,9 @@ trait PathSegmentImpl[A] {
   /** implementation of [[urldsl.errors.PathMatchingError]] for type A. */
   implicit protected val pathError: PathMatchingError[A]
 
-  val root: PathSegment[Unit, A] = PathSegment.root
-  val remainingSegments: PathSegment[List[String], A] = PathSegment.remainingSegments
+  lazy val root: PathSegment[Unit, A] = PathSegment.root
+  lazy val remainingSegments: PathSegment[List[String], A] = PathSegment.remainingSegments
+  lazy val ignoreRemaining: PathSegment[Unit, A] = remainingSegments.ignore(Nil)
   lazy val endOfSegments: PathSegment[Unit, A] = PathSegment.endOfSegments
```

Comment:
Should probably `@deprecate` endOfSegments since there is no more use for it... or is there?

Reply:
Ah yes I wanted to do that, thanks